wandb/util.py

```diff
@@ -390,110 +347,85 @@ def _user_args_to_dict(arguments):
 
 
-def is_pytorch_tensor_typename(typename):
+def is_pytorch_tensor_typename(typename: str) -> bool:
     return typename.startswith("torch.") and (
         "Tensor" in typename or "Variable" in typename
     )
 
 
-def is_jax_tensor_typename(typename):
+def is_jax_tensor_typename(typename: str) -> bool:
     return typename.startswith("jaxlib.") and "DeviceArray" in typename
 
 
-def get_jax_tensor(obj):
-    import jax
+def get_jax_tensor(obj: Any) -> Optional[Any]:
+    import jax  # type: ignore
 
     return jax.device_get(obj)
 
 
-def is_fastai_tensor_typename(typename):
+def is_fastai_tensor_typename(typename: str) -> bool:
     return typename.startswith("fastai.") and ("Tensor" in typename)
 
 
-def is_pandas_data_frame_typename(typename):
+def is_pandas_data_frame_typename(typename: str) -> bool:
     return typename.startswith("pandas.") and "DataFrame" in typename
 
 
-def is_matplotlib_typename(typename):
+def is_matplotlib_typename(typename: str) -> bool:
     return typename.startswith("matplotlib.")
 
 
-def is_plotly_typename(typename):
+def is_plotly_typename(typename: str) -> bool:
     return typename.startswith("plotly.")
 
 
-def is_plotly_figure_typename(typename):
+def is_plotly_figure_typename(typename: str) -> bool:
     return typename.startswith("plotly.") and typename.endswith(".Figure")
 
 
-def is_numpy_array(obj):
+def is_numpy_array(obj: Any) -> bool:
     return np and isinstance(obj, np.ndarray)
 
 
-def is_pandas_data_frame(obj):
+def is_pandas_data_frame(obj: Any) -> bool:
     return is_pandas_data_frame_typename(get_full_typename(obj))
 
 
-def ensure_matplotlib_figure(obj):
+def ensure_matplotlib_figure(obj: Any) -> Any:
     """Extract the current figure from a matplotlib object or return the object if it's a figure.
     raises ValueError if the object can't be converted.
     """
-    import matplotlib
-    from matplotlib.figure import Figure
-
-    # plotly and matplotlib broke in recent releases,
-    # this patches matplotlib to add a removed method that plotly assumes exists
-    from matplotlib.spines import Spine
-
-    def is_frame_like(self):
-        """Return True if directly on axes frame.
-
-        This is useful for determining if a spine is the edge of an
-        old style MPL plot. If so, this function will return True.
-        """
-        position = self._position or ("outward", 0.0)
-        if isinstance(position, str):
-            if position == "center":
-                position = ("axes", 0.5)
-            elif position == "zero":
-                position = ("data", 0)
-        if len(position) != 2:
-            raise ValueError("position should be 2-tuple")
-        position_type, amount = position
-        if position_type == "outward" and amount == 0:
-            return True
-        else:
-            return False
+    import matplotlib  # type: ignore
+    from matplotlib.figure import Figure  # type: ignore
 
-    Spine.is_frame_like = is_frame_like
```

Comment:
Why is this safe to remove.. i dont recall the history, but i know we have had issues in the past here.

Reply:
I tested this on recent plotly/mpl combos and it all worked. You're right, if a user has an old version of plotly, this may cause problems. Let me put it back.

Reply:
awesome. go ahead and merge this once you put this back.  at some point we will have to deal with the mpl/plotly issues, there are a few threads and tickets on some of the issues here. we should capture them in one place sometime.. added to as an SDK refactor task